numpy/f2py/cfuncs.py

```diff
@@ -634,6 +634,9 @@
 fprintf(stderr, "try_pyarr_from_string(str='%s', len=%d, obj=%p)\\n",
         (char*)str,len, obj);
 #endif
+    if (!obj) return -2; /* Object missing */
+    if (obj == Py_None) return -1; /* None */
+    if (!PyArray_Check(obj)) goto capi_fail; /* not an ndarray */
```

Comment:
OK, this looks fine.  I will note that `TRYPYARRAYTEMPLATE` looks terrible, assuming this is correct[^1].  It just assumes anything that isn't an array must be `None` (plus a bit of more madness)?

(seems I hadn't actually clicked on start a review...)


[^1]: it is unclear to me where the heck this is called and thus how the return values are defined/used!)

Reply:
Merged this, but added to an issue (#25200) tracking other things to check within `f2py` sometime